In [ ]:
from google.colab import drive
drive.mount('/content/drive')
%cd /content/drive/MyDrive/workshop3


Mounted at /content/drive
/content/drive/MyDrive/workshop3


In [ ]:
!ls

Titanic-Dataset.csv


In [ ]:
import pandas as pd
import numpy as np

# Problem 1

In [ ]:
df = pd.read_csv('Titanic-Dataset.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [ ]:
fare = df['Fare']
fare.head()

0     7.2500
1    71.2833
2     7.9250
3    53.1000
4     8.0500
Name: Fare, dtype: float64

In [ ]:
class_age = df[['Pclass', 'Age']]
class_age.head()

Pclass   Age
0       3  22.0
1       1  38.0
2       3  26.0
3       1  35.0
4       3  35.0

In [ ]:
survived_gender = df[['Survived', 'Sex']]
survived_gender.head()

Survived     Sex
0         0    male
1         1  female
2         1  female
3         1  female
4         0    male

# Problem 2

In [ ]:
fare_gt_100 = df[df['Fare'] > 100]
fare_gt_100.head()

PassengerId  Survived  Pclass  \
27            28         0       1   
31            32         1       1   
88            89         1       1   
118          119         0       1   
195          196         1       1   

                                               Name     Sex   Age  SibSp  \
27                   Fortune, Mr. Charles Alexander    male  19.0      3   
31   Spencer, Mrs. William Augustus (Marie Eugenie)  female   NaN      1   
88                       Fortune, Miss. Mabel Helen  female  23.0      3   
118                        Baxter, Mr. Quigg Edmond    male  24.0      0   
195                            Lurette, Miss. Elise  female  58.0      0   

     Parch    Ticket      Fare        Cabin Embarked  
27       2     19950  263.0000  C23 C25 C27        S  
31       0  PC 17569  146.5208          B78        C  
88       2     19950  263.0000  C23 C25 C27        S  
118      1  PC 17558  247.5208      B58 B60        C  
195      0  PC 17569  146.5208          B80        C

In [ ]:
first_class = df[df['Pclass'] == 1]
first_class.head()

PassengerId  Survived  Pclass  \
1             2         1       1   
3             4         1       1   
6             7         0       1   
11           12         1       1   
23           24         1       1   

                                                 Name     Sex   Age  SibSp  \
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   
23                       Sloper, Mr. William Thompson    male  28.0      0   

    Parch    Ticket     Fare Cabin Embarked  
1       0  PC 17599  71.2833   C85        C  
3       0    113803  53.1000  C123        S  
6       0     17463  51.8625   E46        S  
11      0    113783  26.5500  C103        S  
23      0    113788  35.5000    A6        S

In [ ]:
female_under_18 = df[(df['Age'] < 18) & (df['Sex'] == 'female')]
female_under_18.head()

PassengerId  Survived  Pclass                                  Name  \
9            10         1       2   Nasser, Mrs. Nicholas (Adele Achem)   
10           11         1       3       Sandstrom, Miss. Marguerite Rut   
14           15         0       3  Vestrom, Miss. Hulda Amanda Adolfina   
22           23         1       3           McGowan, Miss. Anna "Annie"   
24           25         0       3         Palsson, Miss. Torborg Danira   

       Sex   Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
9   female  14.0      1      0   237736  30.0708   NaN        C  
10  female   4.0      1      1  PP 9549  16.7000    G6        S  
14  female  14.0      0      0   350406   7.8542   NaN        S  
22  female  15.0      0      0   330923   8.0292   NaN        Q  
24  female   8.0      3      1   349909  21.0750   NaN        S

In [ ]:
embarked_c_or_s = df[(df['Embarked'] == 'C') | (df['Embarked'] == 'S')]
embarked_c_or_s.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [ ]:
first_second_class = df[(df['Pclass'] == 1) | (df['Pclass'] == 2)]
first_second_class.head()

PassengerId  Survived  Pclass  \
1             2         1       1   
3             4         1       1   
6             7         0       1   
9            10         1       2   
11           12         1       1   

                                                 Name     Sex   Age  SibSp  \
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
9                 Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   

    Parch    Ticket     Fare Cabin Embarked  
1       0  PC 17599  71.2833   C85        C  
3       0    113803  53.1000  C123        S  
6       0     17463  51.8625   E46        S  
9       0    237736  30.0708   NaN        C  
11      0    113783  26.5500  C103        S

In [ ]:
df['Age'] = df['Age'].fillna(df['Age'].median())

In [ ]:
df['Fare_per_year'] = df['Fare'] / df['Age']

In [ ]:
high_fare_age = df[df['Fare_per_year'] > 5]

In [ ]:
high_fare_age_srt = high_fare_age.sort_values(by='Fare_per_year', ascending=False)

In [ ]:
result = high_fare_age_srt[['Name', 'Fare_per_year']]

In [ ]:
result.head()

Name  Fare_per_year
305   Allison, Master. Hudson Trevor     164.728261
297     Allison, Miss. Helen Loraine      75.775000
386  Goodwin, Master. Sidney Leonard      46.900000
164     Panula, Master. Eino Viljami      39.687500
183        Becker, Master. Richard F      39.000000

In [ ]:
df['Fare_per_class'] = df['Fare'] / df['Pclass']

In [ ]:
adult_males = df[(df['Sex'] == 'male') & (df['Age'] >= 18)]

In [ ]:
adult_males_srt = adult_males.sort_values(by='Fare_per_class', ascending=False)

In [ ]:
result = adult_males_srt[['Name', 'Age', 'Fare_per_class']]

In [ ]:
print(result.head())

                                   Name   Age  Fare_per_class
737              Lesurer, Mr. Gustave J  35.0        512.3292
679  Cardeza, Mr. Thomas Drake Martinez  36.0        512.3292
438                   Fortune, Mr. Mark  64.0        263.0000
27       Fortune, Mr. Charles Alexander  19.0        263.0000
118            Baxter, Mr. Quigg Edmond  24.0        247.5208


In [ ]:
total_fare = df['Fare'].sum()

In [ ]:
first_class_fare = df[df['Pclass'] == 1]['Fare'].sum()

In [ ]:
second_class_fare = df[df['Pclass'] == 2]['Fare'].sum()

In [ ]:
third_class_fare = df[df['Pclass'] == 3]['Fare'].sum()

In [ ]:
fare_totals = [first_class_fare, second_class_fare, third_class_fare]

In [ ]:
fare_proportions = [fare / total_fare for fare in fare_totals]

In [ ]:
fare_percentages = [proportion * 100 for proportion in fare_proportions]

In [ ]:
print(f'Percentage of fare revenue by class:')
print(f'First class: {fare_percentages[0]:.2f}%')
print(f'Second class: {fare_percentages[1]:.2f}%')
print(f'Third class: {fare_percentages[2]:.2f}%')

Percentage of fare revenue by class:
First class: 63.35%
Second class: 13.25%
Third class: 23.40%


Percentage of fare revenue by class:
First class: 63.35%
Second class: 13.25%
Third class: 23.40%

In [ ]:
def categorize_age(age):
    if age < 18:
        return 'child'
    elif age < 65:
        return 'adult'
    else:
        return 'senior'

df['Age_group'] = df['Age'].apply(categorize_age)

total_passengers = len(df)

age_group_counts = df['Age_group'].value_counts()

age_group_proportions = age_group_counts / total_passengers

age_group_percentages = age_group_proportions * 100
print('Percentage of passengers by age group:')
print(age_group_percentages)

Percentage of passengers by age group:
Age_group
adult     86.083053
child     12.682379
senior     1.234568
Name: count, dtype: float64


Percentage of passengers by age group:
Age_group

adult     86.083053
child     12.682379
senior     1.234568
Name: count, dtype: float64